pkg/controller/controller_utils.go

```diff
@@ -942,16 +942,6 @@ func PatchNodeTaints(c clientset.Interface, nodeName string, oldNode *v1.Node, n
 		return fmt.Errorf("failed to marshal old node %#v for node %q: %v", oldNode, nodeName, err)
 	}
 
-	newTaints := newNode.Spec.Taints
```

Comment:
Hmmmm well the changes do not impact the oldNode state, since i only deleted : 
 - deepCopy of the oldNode
 - an update of taints on the newNode (newNode already contains the taints)

so i don't see what you mean by  "and is also modified", since that when deepCopying we duplicate objects.